src/java.base/windows/native/libnet/NetworkInterface.c

```diff
@@ -66,720 +56,267 @@ jfieldID ni_ibaddressID;        /* InterfaceAddress.address */
+                setInet6Address_scopeifname(env, inetAddr, netifObj);
+            }
+            bindAddr = (*env)->NewObject(env, ni_ibcls, ni_ibctrID);
+            if (bindAddr == NULL) {
+                freeNetaddrs(addrsHead);
+                return NULL;
+            }
+            (*env)->SetObjectField(env, bindAddr, ni_ibaddressID, inetAddr);
+            if (addrsCurrent->PrefixLength != NO_PREFIX) {
+                (*env)->SetShortField(
+                        env, bindAddr, ni_ibmaskID, addrsCurrent->PrefixLength);
             }
         }
-        (*env)->SetObjectArrayElement(env, addrArr, addr_index, iaObj);
-        (*env)->DeleteLocalRef(env, iaObj);
-        addrs = addrs->next;
-        addr_index++;
+
+        // add the new elements to the arrays
+        (*env)->SetObjectArrayElement(env, addrArr, i, inetAddr);
+        (*env)->DeleteLocalRef(env, inetAddr);
+        (*env)->SetObjectArrayElement(env, bindsArr, i, bindAddr);
+        (*env)->DeleteLocalRef(env, bindAddr);
+
+        // advance to the next address
+        addrsCurrent = addrsCurrent->Next;
+        i++;
     }
-    (*env)->SetObjectField(env, netifObj, ni_addrsID, addrArr);
-    (*env)->SetObjectField(env, netifObj, ni_bindsID, bindsArr);
 
-    free_netaddr(netaddrP);
-    (*env)->DeleteLocalRef(env, name);
-    (*env)->DeleteLocalRef(env, displayName);
+    // free the address list since we no longer need it
+    freeNetaddrs(addrsHead);
+
+    // set the addrs and bindings arrays on the NetworkInterface
+    (*env)->SetObjectField(env, netifObj, ni_addrsID, addrArr);
     (*env)->DeleteLocalRef(env, addrArr);
+    (*env)->SetObjectField(env, netifObj, ni_bindsID, bindsArr);
     (*env)->DeleteLocalRef(env, bindsArr);
 
-    /*
-     * Windows doesn't have virtual interfaces, so child array
-     * is always empty.
-     */
+    // set child array on the NetworkInterface
+    // Windows doesn't have virtual interfaces, so this is always empty
     childArr = (*env)->NewObjectArray(env, 0, ni_class, NULL);
     if (childArr == NULL) {
-      return NULL;
+        return NULL;
     }
     (*env)->SetObjectField(env, netifObj, ni_childsID, childArr);
     (*env)->DeleteLocalRef(env, childArr);
 
-    /* return the NetworkInterface */
     return netifObj;
 }
 
-/*
- * Class:     java_net_NetworkInterface
- * Method:    getByName0
- * Signature: (Ljava/lang/String;)Ljava/net/NetworkInterface;
- */
-JNIEXPORT jobject JNICALL Java_java_net_NetworkInterface_getByName0
-    (JNIEnv *env, jclass cls, jstring name)
-{
-    netif *ifList, *curr;
-    jboolean isCopy;
-    const char *name_utf;
-    jobject netifObj = NULL;
-
-    // Retained for now to support IPv4 only stack, java.net.preferIPv4Stack
-    if (ipv6_available()) {
-        return Java_java_net_NetworkInterface_getByName0_XP (env, cls, name);
-    }
-
-    /* get the list of interfaces */
-    if (enumInterfaces(env, &ifList) < 0) {
+static jobject createNetworkInterfaceForSingleRowWithTables(
```

Comment:
I think it would be helpful for future maintainers if there was a comment on each of static functions to make it clear what they do.